nalgebra-sparse/src/cs.rs

```diff
@@ -679,14 +666,22 @@ where
                 );
 
                 // sort values if they exist
-                if !value_refs.is_empty() {
+                if let Some(values) = values_option.as_mut() {
+                    if values.len() == 0 {
+                        panic!(
+                            "Internal error: Sorting currently not supported if values are empty."
+                        );
+                    }
+                    if minor_indices.len() != values.len() {
+                        panic!("Number of values and minor indices must be the same.");
+                    }
```

Comment:
This is an error condition, not a panic condition. In fact we used to check it [here](https://github.com/dimforge/nalgebra/pull/1015/commits/9af579f6623032b98910e03e048e5c26b33bac97#diff-ad9fa9a189df6658e5fee89093cd2efba6292df75401887d4c3e336a6fca2e4bL563).
I'm a bit puzzled by this change. Why remove the previous check and replace it with this panic? I think the previous check is preferable then.

Reply:
I think you are right, this case could also be tested.

Reply:
And sorry for breaking your link. I rebased current branch and made the link unavailable unintentionally but I have a screenshot of it.
![image](https://user-images.githubusercontent.com/11193727/151443757-b6e127f7-dc30-4a05-8c53-c140dc8023cd.png)